applications/FluidDynamicsApplication/custom_utilities/fluid_mesh_utilities.cpp

```diff
@@ -0,0 +1,130 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "utilities/parallel_utilities.h"
+#include "utilities/reduction_utilities.h"
+
+// Application includes
+#include "fluid_mesh_utilities.h"
+
+namespace Kratos
+{
+
+bool FluidMeshUtilities::AllElementsAreSimplex(const ModelPart& rModelPart)
+{
+    // Check that the provided model part is not empty
+    KRATOS_ERROR_IF(rModelPart.GetCommunicator().GlobalNumberOfElements() == 0) <<
```

Comment:
If you save this in a variable you can skip one communication step

Reply:
Certainly. :+1: